### PR TITLE
Refactor system configuration parsing for improved Destination AE support

### PR DESCRIPTION
**Pull request by yuvraj-wale**
This pull request addresses the comments and suggestions made in the code review.

linked to issue #133.

@pradeeban I feel this should work, if not do let me know, thanks.

### CODE REVIEW

File: modules/cold-extraction/ColdDataRetriever.py [hunk 1]
```diff
@@ -151,33 +151,39 @@ def check_kill_process():
...
+            niffler_processes += 1
+        for _ in os.popen("ps ax | grep -E " + nifflerscp_str + " | grep -v grep"):
+            storescp_processes += 1
+
+        logging.info("Number of running Niffler processes: {0} and storescp processes: {1}".format(niffler_processes,
                                                                                                storescp_processes))
 
-    if niffler_processes > MAX_PROCESSES:
-        logging.error("[EXTRACTION FAILURE] {0}: Previous {1} extractions still running. As such, your extraction "
+        logging.info("{0}: StoreScp process for the current Niffler extraction is starting now".format(
```
Comment (pradeeban) [inline modules/cold-extraction/ColdDataRetriever.py:168]: Hi,
Thanks for your second attempt.

What I am having a hard time figuring out is why did you move this log up?

With this change, now this log will be printed even before the "if niffler_processes > MAX_PROCESSES:" and "if storescp_processes >= niffler_processes:" processes are checked.

Therefore, the log becomes useless (and always printed), whereas previously, it was printed only when it was actually starting the storescp (previous lines 175 - 176).

My suggestions here (and in your previous PR that I closed) are along two lines:
1. Avoid changing things that already work. That should be a separate PR if you have a reason to do so. A PR should be minimal, addressing the problem at hand only.
2. Make sure you don't introduce new bugs while fixing a previous PR.

**Comment by yuvraj-wale** [comment]
oh no that log was not supposed to be there, I am really sorry .Its is supposed to be in the if block above the "if not file_path == cfind_only and not file_path == cfind_detailed" lines of code, sorry Its all little new to me😅. apart from this log change is everything fine? @pradeeban

**Comment by pradeeban** [comment]
Are you a newbie to Python? What are your usual programming languages?

**Comment by yuvraj-wale** [comment]
I primarily use C++ and Dart (flutter) for programming, but I also know some basic Python. I wanted to learn more about this technology and its applications in machine learning and medical science because I am finding it quite fascinating.

**Pull request closed without merging**